### ISSUE TITLE
Evaluating PSMNet on KITTI VO with Pretrained KITTI-2012 model delivers biased and skewed disparity results

### ISSUE THREAD

**Issue by WASCHMASCHINE** [JiaRenChang/PSMNet#113]
I have looked at the KITTI Stereo 2012 dataset, which features parts of the KITTI VO dataset. I conclude that they are part of the same set. This means that I can just use the pretrained 2012 model.
Now, when I evaluate it on sequence 04 of the VO dataset, my disparity values seem to be off compared to ELAS and other stereo algorithms:

![grafik](https://user-images.githubusercontent.com/7250412/53416271-6eb8d380-39d3-11e9-8070-90ccbf492ed7.png)

Also I tried out the other pretrained models. I compared the velodyne projected to the color image. Also, I altered/improved the velodyne to camera calibration, which is somewhat off for the VO dataset.

COMMAND USED: **python submission.py --KITTI 2012 --datapath C:\KittiOdometry\data_odometry_color\dataset\sequences\04\ --loadmodel pretrained_model_KITTI2012.tar**
Note that I altered the KITTI_submission_loader2012.py to work with KITTI VO instead.

Here's my resulting image of the first pair:
![result](https://user-images.githubusercontent.com/7250412/53413979-dff58800-39cd-11e9-9c87-f3713ab4267f.png)

Here are the two input pictures. It would be really nice if somebody would post their disparity results of these two images with the pretrained KITTI 2012 model and hourglass. Then we could have a comparison and identify the source of the error faster.

![left](https://user-images.githubusercontent.com/7250412/53414012-fc91c000-39cd-11e9-8b51-3e9783b48ec9.png)
![right](https://user-images.githubusercontent.com/7250412/53414028-01567400-39ce-11e9-8dcc-d5147c500bca.png)

**Comment by WASCHMASCHINE** [comment]
@JiaRenChang Any advice?

**Comment by JiaRenChang** [comment]
@WASCHMASCHINE 
Hi, I would like to know that the sequence 04 of the VO dataset performs badly?

Here is my testing result, I think it is same as your testing result.
![vo](https://user-images.githubusercontent.com/11732099/53942781-9f48ed80-40f6-11e9-81c4-8dcc2960df7f.png)

**Comment by WASCHMASCHINE** [comment]
@JiaRenChang I think I see no difference between our images.

I decided to evaluate the KITTI2015 model on the KITTI2015 training. It still looks quite off, maybe my evaluation code is wrong? But other disparity results look OK.

![grafik](https://user-images.githubusercontent.com/7250412/54027805-0ae89480-41a3-11e9-8572-c3506c36021c.png)

**Comment by WASCHMASCHINE** [comment]
Here are my 200 disparity images trained with KITTI2015:
`python submission.py --KITTI 2015 --datapath D:\Kitti2015\data_scene_flow_multiview\training\ --loadmodel pretrained_model_KITTI2015.tar`

[Link to .zip](https://ufile.io/bj48h)

It would be great if someone would evaluate them separately, as I am going slowly insane.

**Comment by WASCHMASCHINE** [comment]
I ran the devkit sample from KITTI 2015 and implemented the accuracy measure in my code again.

```
float E = abs(deltaDisparity);
			if (E > 3.0f && E / abs(gtDisparity) > 0.05f){
				++countError;
			}
```

Using the NOC GT I compared the disparity error: They look the same and pretty good (example frame 154: 1.35% accuracy). But I use the same error for my distribution. I conclude that the results of PSMNet are biased, but still very good in terms of KITTI accuracy. Subtracting -1.25 px per pixel from the reconstructed disparities decreased the error even further! @JiaRenChang I would suggest that you may try out evaluating PSMNet with a bias correction.

**Comment by JiaRenChang** [comment]
@WASCHMASCHINE 

Wow, many thanks for your detailed evaluation!
It is interesting that why PSMNet is biased for 1.25 px.
I will try to figure out it.
Thanks again!!

**Comment by zhFuECL** [comment]
@WASCHMASCHINE For KITTI2015, if I trained PSMNet (finetuned from sceneflow dataset) on 160 frames and test on the rest 40 frames,  -1.25 px on disparities cannot get a better result in my experiment.

**Comment by WASCHMASCHINE** [comment]
@zhFuECL Why don't you compute a disparity error histogram like I did?

I used the whole training set of Kitti 2015 to compute the histogram, looked up the bias, and reevaluated it on the training set. I used the pretrained model from the author without modifications for disparities.

**Comment by WASCHMASCHINE** [comment]
@JiaRenChang Any news on the topic?

**Comment by tv12345** [comment]
Hello @WASCHMASCHINE did you use the standard 1242x375 images from the KITTI-2015 testing image folder? or did you scale them down to a lower resolution?
I am trying to get the algorithm working with the pretrained model provided to calculate depth on Jeston TX2.
